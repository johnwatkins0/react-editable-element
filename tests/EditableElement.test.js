import React from 'react';
import renderer from 'react-test-renderer';

import AllDefaults from '../example/src/tests/AllDefaults';
import InitialValue from '../example/src/tests/InitialValue';
import EditableH1 from '../example/src/tests/EditableH1';
import EditableH1WithCustomStyles from '../example/src/tests/EditableH1WithCustomStyles';
import OnChangeOverride from '../example/src/tests/OnChangeOverride';

test('All defaults renders correctly.', () => {
  const tree = renderer.create(<AllDefaults />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Element with an initial value renders correctly.', () => {
  const tree = renderer.create(<InitialValue />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('An editable h1 renders correctly.', () => {
  const tree = renderer.create(<EditableH1 />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('An editable h1 with custom styles renders correctly.', () => {
  const tree = renderer.create(<EditableH1WithCustomStyles />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('An editable with an onchange override renders correctly.', () => {
  const tree = renderer.create(<OnChangeOverride />).toJSON();

  expect(tree).toMatchSnapshot();
});
