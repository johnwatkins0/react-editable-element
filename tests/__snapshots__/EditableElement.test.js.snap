// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All defaults renders correctly. 1`] = `
<input
  className=""
  onBlur={[Function]}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Enter text"
  style={Object {}}
  value=""
/>
`;

exports[`An editable h1 renders correctly. 1`] = `
<h1
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "Click me to edit this text",
    }
  }
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`An editable h1 with custom styles renders correctly. 1`] = `
<h1
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "Click me to edit this text",
    }
  }
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`An editable with an onchange override renders correctly. 1`] = `
<div>
  <input
    className=""
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Enter text"
    style={Object {}}
    value=""
  />
  <div>
    The current value is 
    ""
  </div>
</div>
`;

exports[`Element with an initial value renders correctly. 1`] = `
<div
  className=""
  dangerouslySetInnerHTML={
    Object {
      "__html": "Click me to edit this text.",
    }
  }
  onClick={[Function]}
  style={Object {}}
/>
`;
